engine/orchestrator/score-orchestrator-impl/src/main/java/io/cloudslang/orchestrator/services/SuspendedExecutionCleanerServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright © 2014-2017 EntIT Software LLC, a Micro Focus company (L.P.)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cloudslang.orchestrator.services;
+
+import io.cloudslang.orchestrator.repositories.SuspendedExecutionsRepository;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.PageRequest;
+
+import javax.transaction.Transactional;
+
+
+public class SuspendedExecutionCleanerServiceImpl implements SuspendedExecutionCleanerService {
+
+    private final Integer BULK_SIZE = Integer.getInteger("suspendedexecution.job.bulk.size", 200);
```

Comment:
I can see two problems here:
1. We would need to restart the oo in order for this value to be changed.
  As a solution I would suggest a Spring bean that would hold the variable as/from 'configuration items' and not a system variable. The bean can have a default value in score and another primary bean in oo that gets and updates the item from config
2. higher values for BULK_SIZE could trigger hibernate errors. If a higher value would be configured at the above step then loop through pages of BULK_SIZE and delete one at a time

Reply:
@Razvan-Morar @cristian-nitoiu is a system value, should not be exposed to user. I think it is enough to expose via system property only for extreme cases only.